File: app/src/main/java/com/apppoweron/circularrevealbuttondemo/MainActivity.java
package com.apppoweron.circularrevealbuttondemo;

import android.os.Bundle;
import android.support.v7.widget.Toolbar;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        try {
            loadDefaultFragment();
        } catch (NoContainerException e) {
            e.printStackTrace();
        }
    }

    protected void loadDefaultFragment() throws NoContainerException {
        loadFragment(MainFragment.newInstance(), BaseActivity.FragmentLoadType.ADD);
    }

    @Override
    protected int getContainerId() {
        return R.id.container;
    }
}

File: app/src/main/kotlin/com/apppoweron/circularrevealbuttondemo/MainActivity.kt
package com.apppoweron.circularrevealbuttondemo

import android.os.Bundle
import android.support.v7.widget.Toolbar

class MainActivity : BaseActivity() {

    override val containerId: Int
        get() = R.id.container

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        try {
            loadDefaultFragment()
        } catch (e: NoContainerException) {
            e.printStackTrace()
        }
    }

    @Throws(NoContainerException::class)
    private fun loadDefaultFragment() {
        loadFragment(MainFragment.newInstance(), BaseActivity.FragmentLoadType.ADD)
    }
}
